_posts/2024-07-19-Using-Redis-with-Sidecar.md

```diff
@@ -0,0 +1,179 @@
+            var dataFromDb = GetDataFromDB();
+            await _cache.SetRecordAsync<IEnumerable<WeatherForecast>>(dataFromDb, WeatherData_Key);
+            return dataFromDb;
+        }
+    }
+    ```
+
+4. **Adding Values to the Cache**
+We have an extension method that adds values to the cache:
+
+    ```csharp
+    public static async Task SetRecordAsync<T>(this IDistributedCache cache,
+        T value,
+        string key,
+        TimeSpan? absoluteExpiry = null,
+        TimeSpan? slidingExpiry = null)
+    {
+        // Caching option
+        var cachingOption = new DistributedCacheEntryOptions();
+
+        // Cache Time to Live, from Current Time to <specified> mins, after that cache will expire
+        cachingOption.AbsoluteExpirationRelativeToNow = absoluteExpiry ?? TimeSpan.FromSeconds(60);
+
+        // Cache Time to Live, will slide if it is accessed within the sliding window
+        cachingOption.SlidingExpiration = slidingExpiry;
+
+        // Convert to JSON
+        var dataToStore = JsonSerializer.Serialize(value);
+
+        // Set Cache
+        await cache.SetStringAsync(key, dataToStore, cachingOption);
+    }
+    ```
+
+5. **Running Your Application**
+Now you can run your application. Open your browser and navigate to:
+`http://localhost:5000/weatherforecast`.
+
+The browser will display the weather forecast in JSON format. Since we have implemented logging in the application, you will see logs in the command prompt indicating whether the response is coming from the database or the cache.
+
+## Deploying the Application to Linux App Service
+
+To get started, you'll need to containerize your .NET application. This [tutorial](https://learn.microsoft.com/en-us/dotnet/core/docker/build-container?tabs=windows&pivots=dotnet-8-0) walks you through the process step by step.
+*Note: The Sidecar feature is currently enabled for custom-container scenarios for Linux App Service. We are working on enabling it for code scenarios as well. We will update the blog soon for the code-based web applications*
+
+1. **Containerize Your .NET Application**
+    Create a Dockerfile in the root directory of your project. The Dockerfile for this project is [here](https://github.com/Azure-Samples/sidecar-samples/blob/main/Redis.Core.WebApi-master/Redis.Core.WebApi/Dockerfile).
+2. **Build and Push the Docker Image**
+    Build the image and push it to your preferred container registry, such as Azure Container Registry, Docker Hub, or a private registry.
+
+    ```bash
+    docker build -t <your-registry>/<your-image-name>:<tag> .
+    docker push <your-registry>/<your-image-name>:<tag>
+    ```
+
+    We would also recommend that you push the local redis image to ACR. This would allow you to take advantage of securing your resources using Managed Identity. Please refer to this [documentation](https://learn.microsoft.com/en-us/azure/container-registry/container-registry-get-started-docker-cli?tabs=azure-cli).
```

Comment:
remove */en-us/* from links this help route folks to localized content

Reply:
Done.